### PR TITLE
Add canola plant.

### PR DESCRIPTION
**Pull request by Rail-Runner**
1. Add canola item.
2. Add canola to forage_spring, forage_summer and forage_autumn item groups.
3. Add canola seeds item.
4. Add recipe to make canola seeds.
5. Tweak cooking oil recipe to use canola seeds.
6. Tweak thyme seed name.

Took from Wilderness Overhaul mod by MormonPartyboat.
( http://smf.cataclysmdda.com/index.php?topic=9568 )

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -9352,5 +9352,51 @@
...
+  "autolearn": true,
+  "qualities" : [
+    {"id":"CUT","level":1,"amount":1}
+  ],
+  "components": [
+    [
+      [ "canola", 1 ]
+    ]
+  ]
+},{
```
Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_food.json]: We already have a nearly-identical cooking oil recipe at L:203. Why not just give it ````seed_canola```` as a possible component?

**Comment by Rivet-the-Zombie** [comment]
Cool stuff!

**Comment by kevingranade** [comment]
Hell yea canola!

File: data/json/recipes/recipe_food.json (hunk 1 above)
Comment (Darkyhard) [inline data/json/recipes/recipe_food.json]: Good idea, will do.

**Comment by DeadLeaves** [comment]
This breaks the game. It throws tons of json errors when entering character creation. At least on Windows SDL.

**Comment by BevapDin** [comment]
> It throws tons of json errors when entering character creation.

Of which you quoted none. What are those error you refer to?

**Comment by KA101** [comment]
If so, that's a shame: this looked good in the eyeball test and Darkyhard's been looking more reliable lately.  (Yes, we do pay attention to that sort of thing.)  I'm building a merge now and will fix if I find 'em.

**Comment by sparr** [comment]
there's just one real error, the canola json item is missing a comestible_type. everything after that is json parser wonkery.

**Comment by sparr** [comment]
src/game.cpp:216 Error loading data from json: data/json//items/comestibles.json: line 18677:7: member not found: comestible_type

**Comment by KA101** [comment]
OK, I can have one up in probably half an hour to 45 min.

**Comment by sparr** [comment]
"comestible_type" : "FOOD",

fixes the problem

**Comment by KA101** [comment]
No shit.  Tell the programmer types to lay off their compiler-inattentiveness so I don't have to make clean every night, so perhaps I can build in 10 minutes rather than 30.

**Comment by Darkyhard** [comment]
Oops, sorry for all this mess.

**Pull request merged**